uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,225 @@
+          $httpBackend.flush();
+        });
+
+        it('calls to save() with Task list', function() {
+          var params = {
+            json: {
+              id: TaskMock.taskList,
+              case_id: 1
+            }
+          };
+          requestBody = Task.save.calls.argsFor(0)[1];
+
+          expect(requestBody.json).toEqual(params.json);
+        });
+      });
+
+      describe("when calling TaskService.saveMultiple()", function() {
+        beforeEach(function() {
+          $httpBackend.whenPOST(/action=create_multiple&entity=Task/).respond(function(method, url, data, headers, params) {
+            request.headers = headers;
+            request.data = data;
+
+            return [200];
+          });
+
+          TaskService.saveMultiple(TaskMock.taskList);
+          $httpBackend.flush();
+        });
+
+        it('calls to save() with Task list', function() {
+          var params = {
+            json: {
+              task: TaskMock.taskList
+            }
+          };
+          requestBody = Task.save.calls.argsFor(0)[1],
+
+          expect(requestBody.json).toEqual(params.json);
+        });
+      });
+
+      describe("when calling TaskService.sendReminder()", function() {
+        beforeEach(function() {
+          $httpBackend.whenPOST(/action=sendreminder&entity=Task/).respond(function(method, url, data, headers, params) {
+            request.headers = headers;
+            request.data = data;
+
+            return [200];
+          });
+
+          TaskService.sendReminder(1, TaskMock.reminderNote);
+          $httpBackend.flush();
+        });
+
+        it('calls to save() with Reminder Note', function() {
+          var params = {
+            json: {
+              notes: TaskMock.reminderNote
+            }
+          };
+          requestBody = Task.save.calls.argsFor(0)[1],
+
+          expect(requestBody.json).toEqual(params.json);
+        });
+      });
+    });
+  });
+
+  describe('TaskService', function() {
+    var Task, TaskService, $httpBackend, request = {};;
+
+    beforeEach(module('civitasks.appDashboard'));
+
+    beforeEach(inject(function(_TaskService_, _$httpBackend_) {
+      TaskService = _TaskService_;
+      $httpBackend = _$httpBackend_;
+    }));
+
+    beforeEach(function() {
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+      $httpBackend.whenGET(/action=get&entity=Task/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    describe("save()", function() {
+      beforeEach(function() {
+        $httpBackend.whenPOST(/action=create&entity=Task/).respond(function() {
+          return [200, TaskMock.response.onSave];
+        });
+      });
+
+      it('responds correct Task details', function (){
```

Comment:
i don't undertand what you're testing here, the description is not clear, could you explain it to me?